src/database/database-common/src/transactions/mysql_transaction_manager.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use dill::*;
+use internal_error::{InternalError, ResultIntoInternal};
+use sqlx::MySqlPool;
+
+use crate::{DatabaseTransactionManager, TransactionRef};
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+pub struct MySqlTransactionManager {
+    mysql_pool: MySqlPool,
+}
+
+#[component(pub)]
+#[interface(dyn DatabaseTransactionManager)]
+impl MySqlTransactionManager {
+    pub fn new(mysql_pool: MySqlPool) -> Self {
+        Self { mysql_pool }
+    }
+}
+
+#[async_trait::async_trait]
+impl DatabaseTransactionManager for MySqlTransactionManager {
+    async fn make_transaction(&self) -> Result<TransactionRef, InternalError> {
+        let mysql_transaction = self.mysql_pool.begin().await.int_err()?;
+        Ok(TransactionRef::new(mysql_transaction))
```

Comment:
Should we pass pool instance into `TransactionRef` to open actual DB transaction lazily? This would allow us to avoid any overhead on non-DB related APIs without adding some complex decorators.

Reply:
I'll debug it deeper inside first. It's highly likely the transaction is lazy already within the implementation library. I don't think it actually touche the database connection before executing a first SQL statement.

Reply:
Debugging shows that a connection is reserved and "BEGIN" statement is issued immediately. So, laziness is a desired property if we are to build a more or less universal middlewares.

Implemented `LazyTransactionRef`.